examples/dart/test/read_write_data_test.dart

```diff
@@ -83,4 +83,19 @@ void main() {
     expect(prius!.miles, 500);
     cleanUpRealm(realm);
   });
+
+  test('Write async', () async {
+    final config = Configuration.local([Car.schema]);
+    final realm = Realm(config);
+    // :snippet-start: write-async
+    // Add Subaru Outback to the realm using `writeAsync`
+    Car newOutback = Car("Subaru", model: "Outback Touring XT", miles: 2);
+    await realm.writeAsync(() {
+      realm.add<Car>(newOutback);
+    });
+    // :snippet-end:
+    final outback = realm.find<Car>("Subaru");
+    expect(outback!.miles, 2);
```

Comment:
with the above suggestion accepted, you can test this by seeing that the car has not yet been added to the realm. then subsequently wait for a little while, and see that the car has been populated. 

so smthn like:

```suggestion
    final outback = realm.find<Car>("Subaru");
    expect(outback, isNull);
    expect(realm.isInTransaction, true);
    // let transaction resolve
    await Future.delayed(Duration(milliseconds: 500));
    expect(realm.isInTransaction, false);
    expect(realm.find<Car>("Subaru")!.miles, 2);
```